mmv1/third_party/terraform/services/datastream/resource_datastream_connection_profile_test.go

```diff
@@ -247,3 +359,156 @@ resource "google_datastream_connection_profile" "mysql_con_profile" {
 }
 `, context)
 }
+
+func testAccDatastreamConnectionProfile_sshKey_update(context map[string]interface{}, preventDestroy bool, private_key string, public_key string) string {
+	context["lifecycle_block"] = ""
+	if preventDestroy {
+		context["lifecycle_block"] = `
+        lifecycle {
+            prevent_destroy = true
+        }`
+	}
+	context["private_key"] = private_key
+	context["public_key"] = public_key
+
+	return acctest.Nprintf(`
+resource "random_password" "pwd" {
+	length = 16
+	special = false
+}
+
+resource "google_sql_database_instance" "instance" {
+    depends_on         = [google_compute_instance.default]
+    name            	= "tf-test-my-database-instance%{random_suffix}"
+    database_version	= "POSTGRES_14"
+    region           	= "us-central1"
+    settings {
+        tier = "db-f1-micro"
+        ip_configuration {
+			ipv4_enabled = true
+
+			authorized_networks {
+				value = google_compute_instance.default.network_interface.0.access_config.0.nat_ip
+			}
+        }
+    }
+    
+    deletion_protection  = "false"
+}
+
+resource "google_sql_database" "db" {
+	depends_on = [google_sql_database_instance.instance]
+	instance = google_sql_database_instance.instance.name
+	name     = "db"
+}
+resource "google_sql_user" "user" {
+	depends_on	= [google_sql_database_instance.instance]
+	name		= "user"
+	instance	= google_sql_database_instance.instance.name
+	password	= random_password.pwd.result
```

Comment:
And replace this with something like:

```
password = "password%{random_suffix}"
```

This should fix the VCR tests - it will create a real random password when running the integration test, and use a deterministic password when running the VCR test.